File: jdroid-gradle-commons-plugin/src/main/groovy/com/jdroid/gradle/commons/utils/DefaultPathVisitor.java
package com.jdroid.gradle.commons.utils;

import java.io.IOException;
import java.nio.file.FileVisitResult;
import java.nio.file.FileVisitor;
import java.nio.file.Path;
import java.nio.file.attribute.BasicFileAttributes;

public class DefaultPathVisitor implements FileVisitor<Path> {
	
	@Override
	public FileVisitResult preVisitDirectory(Path path, BasicFileAttributes basicFileAttributes) throws IOException {
		return FileVisitResult.CONTINUE;
	}
	
	@Override
	public FileVisitResult visitFile(Path path, BasicFileAttributes basicFileAttributes) throws IOException {
		return FileVisitResult.CONTINUE;
	}
	
	@Override
	public FileVisitResult visitFileFailed(Path path, IOException e) throws IOException {
		return FileVisitResult.CONTINUE;
	}
	
	@Override
	public FileVisitResult postVisitDirectory(Path path, IOException e) throws IOException {
		return FileVisitResult.CONTINUE;
	}
}

File: jdroid-gradle-commons-plugin/src/main/groovy/com/jdroid/gradle/commons/utils/DefaultPathVisitor.kt
package com.jdroid.gradle.commons.utils

import java.io.IOException
import java.nio.file.FileVisitResult
import java.nio.file.FileVisitor
import java.nio.file.Path
import java.nio.file.attribute.BasicFileAttributes

open class DefaultPathVisitor : FileVisitor<Path> {

    @Throws(IOException::class)
    override fun preVisitDirectory(path: Path, basicFileAttributes: BasicFileAttributes): FileVisitResult {
        return FileVisitResult.CONTINUE
    }

    @Throws(IOException::class)
    override fun visitFile(path: Path, basicFileAttributes: BasicFileAttributes): FileVisitResult {
        return FileVisitResult.CONTINUE
    }

    @Throws(IOException::class)
    override fun visitFileFailed(path: Path, e: IOException): FileVisitResult {
        return FileVisitResult.CONTINUE
    }

    @Throws(IOException::class)
    override fun postVisitDirectory(path: Path, e: IOException): FileVisitResult {
        return FileVisitResult.CONTINUE
    }
}
